taxonomy/utils.py

```diff
@@ -171,14 +274,29 @@ def refresh_product_skills(products, should_commit_to_db, product_type):
     client = EMSISkillsApiClient()
 
     for index, product in enumerate(products, start=1):
+        product = _convert_product_to_dict(product)
+        if product is None:
+            skipped_count += 1
+            continue
         if product_type == ProductTypes.Course:
             skill_attr_val = get_course_metadata_fields_text(skill_extraction_attr, product)
         else:
             skill_attr_val = product[skill_extraction_attr]
         if skill_attr_val:
-            translated_skill_attr = get_translated_skill_attribute_val(
-                product[key_or_uuid], skill_attr_val, product_type
-            )
+            # get metadata of skill_attr_val
+            extra_data = process_skill_attr_text(skill_attr_val, product_type)
```

Comment:
Question: Shouldn't we only call this utility for XBlock type product?

Reply:
The check is done inside the utility to accommodate possibility of additional data for other product types.